Lib/email/message.py

```diff
@@ -1041,7 +1041,13 @@ def iter_attachments(self):
         maintype, subtype = self.get_content_type().split('/')
         if maintype != 'multipart' or subtype == 'alternative':
             return
-        parts = self.get_payload().copy()
+        payload = self.get_payload()
+        # Certain malformed messages can have content type set to `multipart/*`
+        # but still have single part body, in which case payload.copy() can
+        # fail with AttributeError.
+        if not isinstance(payload, list):
```

Comment:
What do you think about catching the `AttributeError` on `.copy()` instead of doing a type test here?  E.g.:

```
        try:
            parts = payload.copy()
        except AttributeError:
            return None
```

?  (Also aside, when returning `None` is part of the API, it's generally better to be explicit.)

Reply:
I'll update to catch `AttributeError`. 

It technically doesn't return `None`, since this is a generator. I don't know if it that makes any sense?

Reply:
It is same what other `return` statements in this method do.

Reply:
Cool, thanks!  I didn't pick that up from the diff context.